skrf/media/tests/test_cpw.py

```diff
@@ -93,9 +115,84 @@ def test_qucs_network(self):
+                ref['n'].plot_s_db(0, 0, ax = axs[0, 0], color = ref['color'])
+                res.plot_s_db(0, 0, ax = axs2[0, 0], linestyle = 'dashed',
+                              color = ref['color'])
+                
+                line.plot_s_deg(0, 0, ax = axs[0, 1], color = ref['color'],
+                               linestyle = 'none', marker = 'x')
+                ref['n'].plot_s_deg(0, 0, ax = axs[0, 1], color = ref['color'])
+                res.plot_s_deg(0, 0, ax = axs2[0, 1], linestyle = 'dashed',
+                              color = ref['color'])
+                
+                line.plot_s_db(1, 0, ax = axs[1, 0], color = ref['color'],
+                               linestyle = 'none', marker = 'x')
+                ref['n'].plot_s_db(1, 0, ax = axs[1, 0], color = ref['color'])
+                res.plot_s_db(1, 0, ax = axs2[1, 0], linestyle = 'dashed',
+                              color = ref['color'])
+                
+                line.plot_s_deg(1, 0, ax = axs[1, 1], color = ref['color'],
+                               linestyle = 'none', marker = 'x')
+                ref['n'].plot_s_deg(1, 0, ax = axs[1, 1], color = ref['color'])
+                res.plot_s_deg(1, 0, ax = axs2[1, 1], linestyle = 'dashed',
+                              color = ref['color'])
+                
+        
+        if self.verbose:
+            axs[1, 0].legend(prop={'size': 6})
+            axs[0, 0].get_legend().remove()
+            axs[0, 1].get_legend().remove()
+            axs[1, 1].get_legend().remove()
+            fig.tight_layout()
+            
+            axs2[1, 0].legend(prop={'size': 6})
+            axs2[0, 0].get_legend().remove()
+            axs2[0, 1].get_legend().remove()
+            axs2[1, 1].get_legend().remove()
+            fig2.tight_layout()
+    
+    def test_ads_network(self):
+        """
+        Test against the ADS simulator results
+        """
+        if self.verbose:
+            fig, axs = plt.subplots(2, 2, figsize = (8,6))
+            fig.suptitle('ads/skrf')
+            fig2, axs2 = plt.subplots(2, 2, figsize = (8,6))
+            fig2.suptitle('ads/skrf residuals')
+            
+        limit_db = 0.1
+        limit_deg = 1.
+        
+        for ref in self.ref_ads:
+            cpw = CPW(frequency = ref['n'].frequency, z0 = 50.,
+                            w = ref['w'], s = ref['s'], t = ref['t'],
+                            h = ref['h'],
+                            has_metal_backside = ref['has_metal_backside'],
+                            ep_r = self.ep_r, rho = self.rho,
+                            tand = self.tand,
+                            compatibility_mode = 'ads',
+                            diel = 'djordjevicsvensson')
+            line = cpw.line(d=self.l, unit='m', embed = True, z0=cpw.Z0)
+            line.name = '`Media.CPW` skrf,ads'
+            
+            # residuals
+            res = line / ref['n']
```

Comment:
I don't think this division is very good for comparing error. When the reference is very close to zero it also requires that the skrf model should also have very small error at that frequency. Usually we care about absolute error with the models and not relative error. I think a better error measure would be just to calculate residual as `line - ref['n']`. Then just the absolute value of error can be checked against a threshold, with possibly different thresholds for S11/S22 and S12/S21. Phase of S21 can be checked separately if you feel like it is important.

Reply:
This makes sense. I did that like a crude deembedding to have the residuals look like a perfect thru and generate plots to document the issue.
However, I could let the "on" demand plot capability but use absolute error to feed the assert.

Reply:
done.